### PR TITLE
Enhancements for Expo Go Compatibility and Type Corrections in Karte SDK for React-Native

### PR DESCRIPTION
**Pull request by naporin0624**
Thank you for crafting the karte SDK for react-native.

## Overview
- Modified for compatibility with Expo Go.
- Corrected some misconfigured return types.

### Details
It was observed that when using Expo Go, `NativeModules.RNKRTXXXModule` becomes undefined. To accommodate for this, I've introduced `undefined` as a union type and employed optional chaining to ensure that API calls are made safely and correctly within the Expo Go environment.

## Additional Notes
- Confirmed that `yarn tsc --noEmit` runs without issues.
- Ensured that `yarn test` executes successfully.
- Verified that `yarn build` operates correctly.
- Created an internal package using `yarn build && npm pack` and validated its functionality within my project.

### CODE REVIEW

File: packages/notification/src/types.tsx [hunk 1]
```diff
@@ -24,5 +24,5 @@ export interface KRTNotificationNativeModule {
 }
 
 export interface RemoteMessage {
-  data?: { [key: string]: string };
+  data?: { [key: string]: string | object };
```
Comment (naporin0624) [inline packages/notification/src/types.tsx:27]: This change is a diff of this PR
https://github.com/plaidev/karte-react-native/pull/58

File: packages/core/src/index.tsx [hunk 2]
```diff
@@ -166,8 +167,8 @@ export class UserSync {
    * ユーザースクリプトとしてWebViewに設定することで、WebView内のタグと連携されます。
    * なおSDKの初期化が行われていない場合はnullを返却します。
    */
-  public static getUserSyncScript(): string {
-    return nativeModule.getUserSyncScript();
+  public static getUserSyncScript(): string | null {
+    return nativeModule?.getUserSyncScript() ?? null;
   }
```
Comment (naporin0624) [inline packages/core/src/index.tsx:172]: > なおSDKの初期化が行われていない場合はnullを返却します。

So the type has been changed to return null.

**Comment by SojiroNishimura** [comment]
@naporin0624 Thank you for contributing our SDK! We have confirmed your PR and discussed about it. In conclusion, it will not be merged as we have found better approach to handle this issue. Please refer to following.

### Summary
* We can reproduce the issue and confirmed nativeModule get to be null on Expo Go environment after executing `expo prebuild --clean`
* It is intended behavior of Expo as the command above builds native codes and they only works for native environment(iOS/Android), not for Expo Go(as it only contains Expo SDK, 3rd party SDKs are not included in it).
* We think adding null checking for native module will hide the difference between native platforms and Expo Go environments, hence developers might get confused as they can not be aware of it.
* It may be better to throw an error at the constructor of KarteCore module if it is running on Expo Go environment (we can check the runtime environment by [this method](https://docs.expo.dev/bare/using-expo-client/#runtime-detection)).

### Conclusion
* That improvement discussed above may be included in the future release but we have fixed our SDK documents so far to inform the environment difference and proper setup steps to avoid the issue.
  * https://developers.karte.io/docs/setup-react-native-sdk-expo#5-%E3%83%8D%E3%82%A4%E3%83%86%E3%82%A3%E3%83%96%E3%82%B3%E3%83%BC%E3%83%89%E3%82%92%E7%94%9F%E6%88%90%E3%81%99%E3%82%8B)

**Comment by naporin0624** [comment]
@SojiroNishimura 

Understood!
I'm using Expo Go and thought that having it work with native modules would enhance the developer experience. However, I see now how behavior that works on Expo Go could lead to confusion with other standard Expo native modules.
I understand why the merge can't happen. Thank you for discussing this.

**Pull request closed without merging**